src/Model/Payment/Factory.php

```diff
@@ -50,15 +46,14 @@ public function __construct($paymentMapping = [])
      */
     public function getPayment(string $paymentMethod): AbstractPayment
     {
-        if (!$this->paymentMethod) {
-            $methodToLoad        = $this->paymentMapping[$paymentMethod] ?? self::DEFAULT_PAYMENT_METHOD;
-            $this->paymentMethod = $this->paymentMapping[strtoupper($methodToLoad)]->create();
-        }
-
-        if (!$this->paymentMethod->isValid()) {
-            $this->paymentMethod = $this->paymentMapping[self::DEFAULT_PAYMENT_METHOD]->create();
-        }
-
-        return $this->paymentMethod;
+        $methodToLoad  = isset($this->paymentMapping[$paymentMethod])
+            ? $paymentMethod
+            : self::DEFAULT_PAYMENT_METHOD;
+        /** @var AbstractPayment $paymentMethodInstance */
+        $paymentMethodInstance = $this->paymentMapping[strtoupper($methodToLoad)]->create();
+
+        return $paymentMethodInstance->isValid()
+            ? $paymentMethodInstance
+            : $this->paymentMapping[self::DEFAULT_PAYMENT_METHOD]->create();
```

Comment:
a bit of a strange one, it's almost a little bit like:
```php
$instance = ($this->paymentMapping[strtoupper($paymentMethod)] ?? $this->paymentMapping[self::DEFAULT_PAYMENT_METHOD])->create();
```
Although I am not sure if it's possible to create it within parentheses like that. You can separate the create part. I think it will have a bit more clarity though as `$methodToLoad` is a bit wonky here.